src/modules/rover_differential/RoverDifferentialControl/RoverDifferentialControl.cpp

```diff
@@ -0,0 +1,178 @@
+			speed_diff_normalized = math::interpolate<float>(speed_diff, -_param_rd_max_speed.get(),
+						_param_rd_max_speed.get(), -1.f, 1.f);
+			speed_diff_normalized = math::constrain(speed_diff_normalized +
+								pid_calculate(&_pid_yaw_rate, _rover_differential_setpoint.yaw_rate_setpoint, vehicle_yaw_rate, 0, dt),
+								-1.f, 1.f); // Feedback
+		}
+
+	} else { // Interpret yaw rate setpoint as already normalized
+		speed_diff_normalized = PX4_ISFINITE(_rover_differential_setpoint.yaw_rate_setpoint) ?
+					math::constrain(_rover_differential_setpoint.yaw_rate_setpoint, -1.f, 1.f) : 0.f;
+	}
+
+	// Speed control
+	float throttle{0.f};
+
+	if (_rover_differential_setpoint.closed_loop_speed
+	    && PX4_ISFINITE(_rover_differential_setpoint.forward_speed_setpoint)) { // Closed loop speed control
+		if (fabsf(_rover_differential_setpoint.forward_speed_setpoint) < FLT_EPSILON) {
+			pid_reset_integral(&_pid_throttle);
+
+		} else {
+			throttle = pid_calculate(&_pid_throttle, _rover_differential_setpoint.forward_speed_setpoint, vehicle_forward_speed, 0,
+						 dt);
+
+			if (_param_rd_max_speed.get() > FLT_EPSILON) { // Feed-forward term
+				throttle += math::interpolate<float>(_rover_differential_setpoint.forward_speed_setpoint,
+								     0.f, _param_rd_max_speed.get(),
+								     0.f, 1.f);
+			}
+		}
+
+	} else { // Interpret speed setpoint as already normalized
+		throttle = PX4_ISFINITE(_rover_differential_setpoint.forward_speed_setpoint) ?
+			   math::constrain(_rover_differential_setpoint.forward_speed_setpoint, -1.f, 1.f) : 0.f;
+	}
+
+	// Publish rover differential status (logging)
+	rover_differential_status_s rover_differential_status{};
+	rover_differential_status.timestamp = _timestamp;
+	rover_differential_status.actual_speed = vehicle_forward_speed;
+	rover_differential_status.actual_yaw_deg = M_RAD_TO_DEG_F * vehicle_yaw;
+	rover_differential_status.desired_yaw_rate_deg_s = M_RAD_TO_DEG_F * _rover_differential_setpoint.yaw_rate_setpoint;
+	rover_differential_status.actual_yaw_rate_deg_s = M_RAD_TO_DEG_F * vehicle_yaw_rate;
+	rover_differential_status.pid_yaw_rate_integral = _pid_yaw_rate.integral;
+	rover_differential_status.pid_throttle_integral = _pid_throttle.integral;
+	rover_differential_status.pid_yaw_integral = _pid_yaw.integral;
+	_rover_differential_status_pub.publish(rover_differential_status);
+
+	// Publish to motors
+	actuator_motors_s actuator_motors{};
+	actuator_motors.reversible_flags = _param_r_rev.get();
+	computeInverseKinematics(throttle, speed_diff_normalized).copyTo(actuator_motors.control);
+	actuator_motors.timestamp = _timestamp;
+	_actuator_motors_pub.publish(actuator_motors);
+
+}
+
+matrix::Vector2f RoverDifferentialControl::computeInverseKinematics(float forward_speed_normalized,
+		const float speed_diff_normalized)
+{
+	float combined_velocity = fabsf(forward_speed_normalized) + fabsf(speed_diff_normalized);
+
+	if (combined_velocity > 1.0f) { // Prioritize yaw rate
```

Comment:
Add comment where the >1 threshold comes from (what physical meaning it has)

Reply:
I just meant adding a comment like "prioritize yaw if commanded combined velocity exceeds (normalized) limit of 1"

Reply:
Implemented [here](https://github.com/PX4/PX4-Autopilot/pull/23629/commits/2c76eb214a78a493dc9e1aac76e988cf7a0ec2b3#diff-dddbbc48e8457e4320c90a590999b898914d2233d4757bdbff320d8a87ce2b45R166-R170).